smithy4s-client/src/main/scala/kinesis4cats/smithy4s/client/KinesisClient.scala

```diff
@@ -0,0 +1,109 @@
+import cats.effect.{Async, Resource}
+import cats.syntax.all._
+import com.amazonaws.kinesis._
+import org.http4s.client.Client
+import org.http4s.{Request, Response}
+import org.typelevel.log4cats.StructuredLogger
+import org.typelevel.log4cats.noop.NoOpLogger
+import smithy4s.aws._
+import smithy4s.aws.http4s._
+
+import kinesis4cats.logging._
+import kinesis4cats.smithy4s.client.middleware.RequestResponseLogger
+
+/** A thin wrapper around
+  * [[https://disneystreaming.github.io/smithy4s/docs/overview/intro/ Smithy4s's]]
+  * [[https://disneystreaming.github.io/smithy4s/docs/protocols/aws/aws AWS]]
+  * protocol.
+  *
+  * Per Smithy4s's documentation, this is an experimental module that is not
+  * expected to be production ready.
+  *
+  * Updates to the smithy file(s) in this module are not intended to be
+  * backwards compatible.
+  *
+  * [[https://docs.aws.amazon.com/kinesis/latest/APIReference/API_SubscribeToShard.html SubscribeToShard]]
+  * is not a currently supported operation.
+  */
+object KinesisClient {
+
+  /** Helper class containing required
+    * [[kinesis4cats.logging.LogEncoder LogEncoders]] for the
+    * [[kinesis4cats.smithy4s.client.KinesisClient KinesisClient]]
+    */
+  final class LogEncoders[F[_]](implicit
+      val http4sRequestLogEncoder: LogEncoder[Request[F]],
+      val http4sResponseLogEncoder: LogEncoder[Response[F]]
+  )
+
+  /** Create a KinesisClient [[cats.effect.Resource Resource]]
+    *
+    * @param client
+    *   [[https://http4s.org/v0.23/docs/client.html Client]] implementation for
+    *   the api calls
+    * @param region
+    *   [[https://github.com/disneystreaming/smithy4s/blob/series/0.17/modules/aws-kernel/src/smithy4s/aws/AwsRegion.scala AwsRegion]]
+    *   in use
+    * @param loggerF
+    *   [[cats.effect.Async Async]] => [[cats.effect.Async Async]] of
+    *   [[https://github.com/typelevel/log4cats/blob/main/core/shared/src/main/scala/org/typelevel/log4cats/StructuredLogger.scala StructuredLogger]].
+    *   Default is
+    *   [[https://github.com/typelevel/log4cats/blob/main/noop/shared/src/main/scala/org/typelevel/log4cats/noop/NoOpLogger.scala NoOpLogger]]
+    * @param credsF
+    *   [[https://github.com/disneystreaming/smithy4s/blob/series/0.17/modules/aws/src/smithy4s/aws/SimpleHttpClient.scala SimpleHttpClient]]
+    *   \=> [[smithy4s.aws.AwsCredentials AwsCredentials]]. Default to
+    *   [[https://github.com/disneystreaming/smithy4s/blob/series/0.17/modules/aws/src/smithy4s/aws/AwsCredentialsProvider.scala AwsCredentialsProvider.default]]
+    * @param F
+    *   [[cats.effect.Async Async]]
+    * @param LE
+    *   [[kinesis4cats.smithy4s.client.KinesisClient.LogEncoders KinesisClient.LogEncoders]]
+    * @return
+    *   [[cats.effect.Resource Resource]] of a Kinesis Client.
+    */
+  def apply[F[_]](
+      client: Client[F],
+      region: AwsRegion,
+      loggerF: Async[F] => F[StructuredLogger[F]] = (f: Async[F]) =>
+        f.pure(NoOpLogger[F](f)),
```

Comment:
Default of NoOpLogger, allows users to configure their own backend for the logging.